samplers/parser.go

```diff
@@ -569,11 +566,29 @@ func ParseServiceCheck(packet []byte) (*UDPServiceCheck, error) {
 				return nil, errors.New("Invalid service check packet, multiple tag sections")
 			}
 			tags := strings.Split(string(pipeSplitter.Chunk()[1:]), ",")
-			ret.Tags = tags // no need to sort, we don't aggregate on this
+			mappedTags := ParseTagSliceToMap(tags)
+			for k, v := range mappedTags {
+				ret.Tags[k] = v
+			}
 		default:
 			return nil, errors.New("Invalid service check packet, unrecognized metadata section")
 		}
 	}
 
 	return ret, nil
 }
+
+// ParseTagSliceToMap handles splitting a slice of string tags on `:` and
+// creating a map from the parts.
+func ParseTagSliceToMap(tags []string) map[string]string {
```

Comment:
so, tangent! One thing that I have been thinking about is that `veneursinkonly` tag that can have multiple occurrences with the same key and multiple values. I wonder if there's a representation we can use that still allows users to do that & translates do SSF correctly.

Reply:
Eep! Good question. That get us into `map[string][]string` territory like HTTP headers or something… I don't think I can solve it here but that's a great thing for us to figure out.